internal/cephfs/controllerserver.go

```diff
@@ -60,7 +60,7 @@ func (cs *ControllerServer) createBackingVolume(
 	if sID != nil {
 		if err = cs.OperationLocks.GetRestoreLock(sID.SnapshotID); err != nil {
 			util.ErrorLog(ctx, err.Error())
-			return status.Error(codes.Aborted, err.Error())
+			return fmt.Errorf("failed to get restore lock: %w", status.Error(codes.Aborted, err.Error()))
```

Comment:
This might actually be problematic. The `status.Error()` returns a grpc-error, not sure if (or how) that is retained when it is re-formatted with `%w`. I really dislike it that some functions return grpc-errors, and others don't. IMHO only CSI related calls should return grpc-errors, and helper functions should stick to 'normal' error. Or at least helpers that return grpc-errors should have a comment at the top of the function about it.

Reply:
I think for now we can disable this static check and we can bring it back once we address returning proper error messages handling from internal and outer functions. @nixpanic any thought?

Reply:
@Madhu-1  @nixpanic We have many other similar changes. So shall I disable this linter or make further changes?
IMO we can merge the changes that don't have `status.Error()` and disable the linter. Later on, when we have proper error messages, we can work on this. If you want I can open an issue for the same or add TODO while disabling the linter.
Let me know how shall I proceed with this.

Reply:
The `status.Error*()` calls should only be done in the `controllerserver.go` and `nodeserver.go` files, I think. If there is an annotation like `// nolint:...` that you can place in the top of those files, and enable the linter for everything else, that would be my preference.

You might need to cleanup some of the error returning, as I suspect that some other files return gRPC errors with `status.Error*()` too, those should become normal errors, and the calling function should convert them to gRPC errors. This might not be a trivial change though. Place such changes in their own commit (per function, or group of functions) and maybe even their own PRs.

Reply:
@nixpanic as there are other corner cases to be considered as well, disabling this for now and will add a TODO.